giraffe/src/utils/escapeCSVFieldWithSpecialCharacters.ts

```diff
@@ -0,0 +1,13 @@
+export const wrapInDoubleQuotes = (str: string): string => {
+  return `"${str}"`
+}
```

Comment:
is there any indication that this convenience function will contribute to memory issues? it will create a single variable `str` which stays in scope while the function is being called, and then is re-allocated.

Reply:
Yes, in fact a good amount of the function duplication that has been removed in the parser thus far contributed to a spike in memory allocation, which in term reduced the performance of the parser. Given that a parser should be able to handle 28mb of CSV, and given that we might encounter a CSV that meets this criteria at every single line of table data, it would be prudent for us to minimize readability for performance

Reply:
I was hoping for a little bit more information from you. "Memory allocation issues" and "reduced the performance" are quite vague because they could refer to any number of different things. What specific issues are you seeing? Is the parser running out of memory? Is garbage collection affecting performance? Is parsing taking too long? In what _specific_ way does this convenience function affect that?

Reply:
The parser duplicates memory for every iteration of the chunk. Memory allocation issues are affecting performance.